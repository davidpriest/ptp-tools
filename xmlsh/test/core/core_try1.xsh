# try/catch/finally basics

#
# several lexical forms of try with no failures
#
try { true ; } catch X { echo Failed Catch ; } finally { echo Success ; }

try 
{ 
	true ; 
} catch X 
{ 
	echo Failed Catch  
} finally 
{ 
echo Success ; 
}

#
# try with no finally, several forms 
#
try { true ; } 
catch X { echo Failed catch ; }
echo Success No Finally 

try { true ; } catch X { echo Failed catch ; }
echo Success No Finally 

try 
{ true ; } 
catch X 
{ 
	echo Failed catch ; 
}
echo Success No Finally 


# Throw a string

try {
	echo Before throw
	throw "A string"
	echo Fail SNH after thorw
} catch X
{
	echo Catch success: $X
}



