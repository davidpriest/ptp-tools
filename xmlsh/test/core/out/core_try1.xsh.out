Success
Success
Success No Finally
Success No Finally
Success No Finally
Before throw
Catch success: A string
